### ISSUE TITLE
自定义metric不生效

### ISSUE THREAD

**Issue by Hspix** [modelscope/ms-swift#4372]
**Describe the bug**
参考[metric自定义](https://swift.readthedocs.io/zh-cn/latest/Customization/%E6%8F%92%E4%BB%B6%E5%8C%96.html)以及在[metric.py](https://github.com/modelscope/ms-swift/blob/main/swift/plugin/metric.py#L132)中添加自定义metric，如下
```
def compute_metric(eval_prediction: EvalPrediction) -> Dict[str, float]:
    return {"test_acc": 1}


# Add your own metric calculation method here, use --metric xxx to train
METRIC_MAPPING = {
    #'acc': (compute_acc_metrics, preprocess_logits_for_acc),
    #'nlg': (compute_nlg_metrics, None),
    'test_acc': (compute_metric,preprocess_logits_for_acc)
}
```

执行以下脚本
```
CUDA_VISIBLE_DEVICES=0,1 \
swift sft \
    --model /home/models/qwen/2.5/coder/3b/instruct \
    --model_type qwen2_5 \
    --train_type lora \
    --dataset 'data.train.json#1' \
    --split_dataset_ratio 0 \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --learning_rate 1e-4 \
    --warmup_ratio 0.05 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --eval_steps 50 \
    --save_steps 100 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --max_length 4096 \
    --output_dir output \
    --temperature 0.9 \
    --metric test_acc \
    --dataloader_num_workers 4
```

但在后续日志内容中找不到`test_acc`，
```
[INFO:swift] model_parameter_info: PeftModelForCausalLM: 3100.9055M Params (14.9668M Trainable [0.4827%]), 0.0001M Buffers.
/usr/local/lib/python3.11/site-packages/swift/trainers/mixin.py:89: FutureWarning: `tokenizer` is deprecated and will be removed in version 5.0.0 for `Seq2SeqTrainer.__init__`. Use `processing_class` instead.
  super().__init__(
[2025-05-27 11:54:18,694] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/usr/local/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py:47: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @autocast_custom_fwd
/usr/local/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py:66: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @autocast_custom_bwd
[INFO:swift] The logging file will be saved in: /output/v14-20250527-115413/logging.jsonl
{'loss': 0.40608531, 'token_acc': 0.85365854, 'grad_norm': 0.90274876, 'learning_rate': 0.0001, 'memory(GiB)': 8.56, 'train_speed(iter/s)': 0.585343, 'epoch': 1.0, 'global_step/max_steps': '1/1', 'percentage': '100.00%', 'elapsed_time': '1s', 'remaining_time': '0s'}
Train: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.41s/it][INFO:swift] Saving model checkpoint to /output/v14-20250527-115413/checkpoint-1
{'train_runtime': 1.8372, 'train_samples_per_second': 0.544, 'train_steps_per_second': 0.544, 'train_loss': 0.40608531, 'epoch': 1.0, 'global_step/max_steps': '1/1', 'percentage': '100.00%', 'elapsed_time': '1s', 'remaining_time': '0s'}
```

**Your hardware and system info**
`ms_swift                          3.4.1.post1`
